### PR TITLE
Invite institution email html

### PR DESCRIPTION
**Pull request by raonismaneoto**
<p><b>Feature/Bug description:</b>
Create invite institution email sender and its html.
</p>

<p><b>Solution:</b>
I've created the html that is pretty much similar to invite_user.html and extended the email_sender hierarchy to handle this case.
</p>

<p><b>TODO/FIXME:</b> n/a</p>

### CODE REVIEW

File: backend/models/invite.py [hunk 1]
```diff
@@ -71,17 +71,16 @@ def sendInvite(self, user, host):
 
     def send_email(self, host, receiver_email, body=None):
         """Method of send email of invite user."""
-        body = body or """Você foi convidado a participar da plataforma e-CIS,
+        body = body or """Você foi convidado a participar da plataforma CIS,
```
Comment (andreldsa) [inline backend/models/invite.py]: Extract those string to a separated language file.

File: backend/send_email_hierarchy/email_sender.py [hunk 2]
```diff
@@ -24,6 +24,10 @@ def send_email(self, email_json=None):
                 'invitee': self.receiver,
                 'subject': self.subject,
                 'html': self.html,
-                'json': json.dumps(email_json if email_json else self.body)
+                'json': json.dumps(email_json if email_json else {'body': self.body})
             }
         )
+
+    def crop_institution_name(self, institution_name):
+        institution_name = (institution_name[:26] + '...') if len(institution_name) > 29 else institution_name
```
Comment (andreldsa) [inline backend/send_email_hierarchy/email_sender.py]: Magic numbers

File: backend/send_email_hierarchy/email_sender.py [hunk 3]
```diff
@@ -13,7 +13,7 @@ def __init__(self, **kwargs):
         self.subject = kwargs['subject']
         self.receiver = kwargs['receiver']
         self.html = 'default.html'
-        self.body = ""
+        self.body = kwargs['body'] if 'body' in kwargs.keys() else ""
 
     def send_email(self, email_json=None):
         taskqueue.add(
```
Comment (andreldsa) [inline backend/send_email_hierarchy/email_sender.py]: Missing docstring

File: backend/send_email_hierarchy/email_sender.py [hunk 4]
```diff
@@ -24,6 +24,10 @@ def send_email(self, email_json=None):
                 'invitee': self.receiver,
                 'subject': self.subject,
                 'html': self.html,
-                'json': json.dumps(email_json if email_json else self.body)
+                'json': json.dumps(email_json if email_json else {'body': self.body})
             }
         )
+
+    def crop_institution_name(self, institution_name):
```
Comment (andreldsa) [inline backend/send_email_hierarchy/email_sender.py]: Missing docstring

File: backend/send_email_hierarchy/invite_institution_email_sender.py [hunk 5]
```diff
@@ -0,0 +1,21 @@
+"""Invite institution email sender model."""
+
+from email_sender import EmailSender
+
+
+class InviteInstitutionEmailSender(EmailSender):
+
+    def __init__(self, **kwargs):
```
Comment (andreldsa) [inline backend/send_email_hierarchy/invite_institution_email_sender.py]: Missing docstring

File: backend/send_email_hierarchy/invite_institution_email_sender.py [hunk 6]
```diff
@@ -0,0 +1,21 @@
+"""Invite institution email sender model."""
+
+from email_sender import EmailSender
+
+
+class InviteInstitutionEmailSender(EmailSender):
```
Comment (andreldsa) [inline backend/send_email_hierarchy/invite_institution_email_sender.py]: Missing docstring

File: backend/send_email_hierarchy/invite_institution_email_sender.py [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+
+    def __init__(self, **kwargs):
+        super(InviteInstitutionEmailSender, self).__init__(**kwargs)
+        self.html = 'invite_institution_email.html'
+        self.inviter = kwargs['inviter']
+        self.institution = self.crop_institution_name(kwargs['institution'])
+        self.invited_institution = self.crop_institution_name(kwargs['invited_institution'])
+
+    def send_email(self):
```
Comment (andreldsa) [inline backend/send_email_hierarchy/invite_institution_email_sender.py]: Missing docstring

File: backend/send_email_hierarchy/invite_user_email_sender.py [hunk 8]
```diff
@@ -8,7 +8,7 @@ def __init__(self, **kwargs):
         super(InviteUserEmailSender, self).__init__(**kwargs)
         self.html = 'invite_user.html'
         self.inviter = kwargs['inviter']
-        self.institution = kwargs['institution']
+        self.institution = self.crop_institution_name(kwargs['institution'])
 
     def send_email(self):
```
Comment (andreldsa) [inline backend/send_email_hierarchy/invite_user_email_sender.py]: Missing docstring

File: backend/models/invite.py [hunk 9]
```diff
@@ -69,19 +70,13 @@ def sendInvite(self, user, host):
...
+    def send_email(self, host, receiver_email, body):
         """Method of send email of invite user."""
-        body = body or """Você foi convidado a participar da plataforma e-CIS,
-        para realizar o cadastro acesse http://%s
-
-        Equipe e-CIS
-        """ % (host)
-        subject = "Convite plataforma e-CIS"
-        email_sender = InviteUserEmailSender(**{
+        subject = get_common_string('INVITE_EMAIL_SUBJECT')
```
Comment (andreldsa) [inline backend/models/invite.py]: Why not use directly the resource? 

`subject = strings.INVITE_EMAIL_SUBJECT`

File: backend/send_email_hierarchy/invite_institution_email_sender.py [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+
+
+class InviteInstitutionEmailSender(EmailSender):
+    """Entity responsible for send invite institution's email."""
+
+    def __init__(self, **kwargs):
+        """The class constructor.
+
+        It initializes the object with its html and its specific properties.
+        crop_institution_name is called to get sure that the names don't exceed the maximum allowed size.
```
Comment (andreldsa) [inline backend/send_email_hierarchy/invite_institution_email_sender.py]: to make sure*

File: backend/send_email_hierarchy/invite_user_email_sender.py [hunk 11]
```diff
@@ -3,14 +3,24 @@
 from email_sender import EmailSender
 
 class InviteUserEmailSender(EmailSender):
+    """Entity responsible for send invite user's email."""
 
     def __init__(self, **kwargs):
+        """The class constructor.
+
+        It initializes the object with its html and its specific properties.
+        crop_institution_name is called to get sure that the names don't exceed the maximum allowed size.
```
Comment (andreldsa) [inline backend/send_email_hierarchy/invite_user_email_sender.py]: to make sure*

File: backend/util/strings_pt_br.py [hunk 12]
```diff
@@ -0,0 +1,13 @@
+# -*- coding: utf-8 -*-
+"""This file encapsulates all the system's pt-br strings."""
+
+
+STRINGS = {
```
Comment (andreldsa) [inline backend/util/strings_pt_br.py]: You can use this directly from the file, like:

```
from util.strings_pt_br import STRINGS

STRINGS['INVITE_EMAIL_SUBJECT']
```

Comment (raonismaneoto) [reply backend/util/strings_pt_br.py]: Yes I can, but what if there were many strings into STRINGS that need complement? I would have to import like you said and import the method. If I import only the file, I would call the method with util.strings_pt_br.get_string(). What you think?

**Pull request merged**